unittests/gettestkernels.py

```diff
@@ -0,0 +1,307 @@
+"""
```

Comment:
I think this file shouldn't be in here at all, right? You renamed it `kernelcache.py` and use that in your later imports. Because this file can't even work as written (see `cwd=` for example).